chirp/drivers/tdh8.py

```diff
@@ -0,0 +1,2513 @@
+        elif mode == 'Tone':
+            #memval.set_value(int(value * 10))
+            val = str(int(value * 10)).rjust(4, '0')
+            # stx = '%02X' % val
+            stx_f = str(val)[2:]
+            stx_b = str(val)[:2]
+            memval.set_value([int(stx_f, 16), int(stx_b, 16)])
+            
+        elif mode == 'DTCS':
+            val = str(value).rjust(3, '0')
+            if pol == 'R':
+                memval.set_value([int(str(val[1:]), 16), int('C' + str(val[0]), 16)])
+            elif pol == 'N':
+                memval.set_value([int(str(val[1:]), 16), int('8' + str(val[0]), 16)])
+        else:
+            raise Exception("Internal error: invalid mode `%s'" % mode)
+
+    def _get_mem(self, number):
+        return self._memobj.memory[number]
+
+    def _get_nam(self, number):
+        return self._memobj.names[number]
+    
+    def _get_fm(self, number):
+        return self._memobj.fmmode[number]
+    
+    def _get_get_scanvfo(self, number):
+        return self._memobj.fmvfo[number]
+    
+    def _get_scan(self, number):
+        return self._memobj.scanadd[number]
+
+    def _get_block_data(self, number):
+        return self._memobj.blockfirm[number]
+
+    def _get_fmblock_data(self, number):
+        return self._memobj.fmblockfirm[number]
+
+    def get_memory(self, number):
+        _mem = self._get_mem(number)
+        _nam = self._get_nam(number)
+                
+        mem = chirp_common.Memory()
+        
+        mem.number = number
+
+        if _mem.get_raw()[0] == "\xff":
+            mem.empty = True
+            return mem
+        
+        # Receiving frequency
+        mem.freq = int(_mem.rxfreq) * 10
+
+        # Offset
+        if self._gmrs:
+            mem.duplex = ""
+            mem.offset = 0
+        else:
+            if self._is_txinh(_mem):
+                mem.duplex = ""
+                mem.offset = 0
+            elif int(_mem.rxfreq) == int(_mem.txfreq):
+                mem.duplex = ""
+                mem.offset = 0
+            elif abs(int(_mem.rxfreq) * 10 - int(_mem.txfreq) * 10) > 70000000:
+                mem.duplex = "split"
+                mem.offset = int(_mem.txfreq) * 10
+            else:
+                mem.duplex = int(_mem.rxfreq) > int(_mem.txfreq) and "-" or "+"
+                mem.offset = abs(int(_mem.rxfreq) - int(_mem.txfreq)) * 10
+
+        # Channel name
+        for char in _nam.name:
+            if str(char) == "\xFF":
+                char = " "  # The UV-5R software may have 0xFF mid-name
+            mem.name += str(char)
+        mem.name = mem.name.rstrip()
+        if (MODE_RADIO == "P31185" or MODE_RADIO == "P31184") and (mem.number >= 189 and mem.number <=199):
+            mem.name = HAM_NAME[mem.number - 200]
+
+        # dtcs_pol = ["N", "N"]
+        
+        # encode and decode
+        # dec
+        lin2 = '%02X' % _mem.rxtone[1] + '%02X' % _mem.rxtone[0]
+        rxtone = self._decode_tone(lin2)
```

Comment:
Ok, I'll fix it to the next commit

Reply:
I found it easier to identify Tmode with u8 rxtone[2], because I can get the header identification, trying to use lbcd to deal with integers will be more complicated.

Reply:
You're literally doing the same thing though, but converting to strings and then parsing the strings back to integers.

Please, use bcd and integers.